src/DataMapper/Product/PopularityDataMapper.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\DataMapper\Product;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Setono\Doctrine\ORMTrait;
+use Setono\SyliusMeilisearchPlugin\DataMapper\DataMapperInterface;
+use Setono\SyliusMeilisearchPlugin\Document\Document;
+use Setono\SyliusMeilisearchPlugin\Document\Product;
+use Setono\SyliusMeilisearchPlugin\Model\IndexableInterface;
+use Setono\SyliusMeilisearchPlugin\Provider\IndexScope\IndexScope;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Product\Model\ProductVariantInterface;
+use Webmozart\Assert\Assert;
+
+final class PopularityDataMapper implements DataMapperInterface
```

Comment:
For the first try, it's 👍 but I wonder if it wouldn't be too heavy regarding the database connections in larger shops. Maybe, we should introduce an `int $boughtCount` on the `Product` entity, which would be updated after the order is paid. This way, we have a single scalar value to synchronize into the mailisearch while indexing (no additional big queries with joins needed). And such an operation (raising the counter on the product) does not seem complicated or demanding at all 🖖

Reply:
I agree and I am not the biggest fan of this either. I _did_ think about your approach though and I am a bit reluctant to adding more properties to entities if it's not strictly needed. Also a problem would arise with that approach and that is if we just increment an integer, we would get _all time_ sales and not the sales the last three months.

I just did the query on a orders table where I have around 15,000 orders / 30 days and the query took 38 ms.